scripts/infra/cloud-instance.sh

```diff
@@ -0,0 +1,400 @@
+    python3 -c "import sys, json; print(json.load(sys.stdin)['network_interfaces'][0]['floating_ips'][0]['address'])")
+}
+
+instance_user_name() {
+    local cloud_type=$1
+    if [ "$cloud_type" = 'ec2' ]; then
+        echo "ec2-user"
+    elif [ "$cloud_type" = 'ibm' ]; then
+        echo "root"
+    fi
+}
+
+instance_user_home() {
+    local cloud_type=$1
+    if [ "$cloud_type" = 'ec2' ]; then
+        echo "/home/ec2-user"
+    elif [ "$cloud_type" = 'ibm' ]; then
+        echo "/root"
+    fi
+}
+
+setup_rh_devenv() {
+    local cloud_type=$1
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf update -y
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf install git gcc make pip python3 python3-devel -y
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" "sudo dnf install g++ -y || sudo dnf install gcc-c++"
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" "if [ ! -d instructlab.git ]; then git clone --bare https://github.com/instructlab/instructlab.git && git clone instructlab.git && pushd instructlab && git remote add syncrepo ../instructlab.git; fi "
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" "pushd instructlab && python3 -m venv --upgrade-deps venv"
+}
+
+pip_install_with_nvidia() {
+    local cloud_type=$1
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" "pushd instructlab && source venv/bin/activate && pip cache remove llama_cpp_python && pip install . -C cmake.args='-DLLAMA_CUBLAS=on' && ilab"
+}
+
+pip_install_with_amd() {
+    local cloud_type=$1
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" "pushd instructlab && source venv/bin/activate && pip cache remove llama_cpp_python && \
+    pip install instructlab --extra-index-url https://download.pytorch.org/whl/rocm6.0 \
+   -C cmake.args='-DLLAMA_HIPBLAS=on' \
+   -C cmake.args='-DAMDGPU_TARGETS=all' \
+   -C cmake.args='-DCMAKE_C_COMPILER=/opt/rocm/llvm/bin/clang' \
+   -C cmake.args='-DCMAKE_CXX_COMPILER=/opt/rocm/llvm/bin/clang++' \
+   -C cmake.args='-DCMAKE_PREFIX_PATH=/opt/rocm'"
+}
+
+install_rh_nvidia_drivers() {
+    local cloud_type=$1
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" "sudo dnf install kernel-devel-\$(uname -r) kernel-headers-\$(uname -r) -y"
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf install https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm -y
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel9/x86_64/cuda-rhel9.repo
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf clean expire-cache
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf module install nvidia-driver:latest-dkms -y
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf install cuda-toolkit nvidia-gds -y
```

Comment:
FYI this is a more comprehensive list of packages I installed to get vllm working as a serving backend. 
```shell
sudo dnf install -y cuda-toolkit nvidia-gds libnccl-devel libcudnn8 nvidia-driver-NVML nvidia-driver-cuda-libs`
```

I can open up another PR with this if you'd rather not include it in this one.

Reply:
I went ahead and included.  Thanks!